src/core/include/ngraph/log.hpp

```diff
@@ -70,7 +70,11 @@ constexpr const char* get_file_name(ConstString s) {
 }
 NGRAPH_API_DEPRECATED
 constexpr const char* trim_file_name(ConstString root, ConstString s) {
+#ifdef PROJECT_ROOT_DIR_LENGTH
```

Comment:
it's a public header.
when samples are compiled against prebuilt OpenVINO, this macro is not defined. So, should we move implementation into .cpp ?